### ISSUE TITLE
Support per-module disables

### ISSUE THREAD

**Issue by eddie-dunn** [PyCQA/pylint#871]
I would like to disable a set of warnings for modules in a specific 
folder.

Why? We have a Python project laid out like this:

```
    project/
    ├── .pylintrc
    ├── module1.py
    ├── module2.py
    └── tests/
        ├── test_module1.py
        └── test_module2.py
```

Our unit and function tests reside in the folder called `tests/`. When it comes to tests the pylint warnings `missing-docstring`, `invalid-name` and `protected-access` are not relevant. On the other hand, these warnings are very useful for the actual code in the project.

Also, we would like to avoid adding #-disable boilerplate inside every single file; preferably we would like to add the folder-specific disables in the project's pylintrc-file. E.g, something along the lines of

```
 folder-disabled=tests/:invalid-name,protected-access
```

in the pylintrc.

**Comment by kernc** [comment]
Blocked on https://github.com/PyCQA/pylint/issues/618 and https://github.com/PyCQA/pylint/issues/175#issuecomment-182689321.

**Comment by AWhetter** [comment]
As @kernc mentioned, this is a duplicate of #618. It's actively being worked on so should be solved soon!
